contrib/filter-repo-demos/timeshift.py

```diff
@@ -0,0 +1,55 @@
+#!/usr/bin/env python3
+
+import argparse
+import os
+import subprocess
+import datetime
+import sys
+try:
+  import git_filter_repo as fr
+except ImportError:
+  raise SystemExit("Error: Couldn't find git_filter_repo.py.  Did you forget to make a symlink to git-filter-repo named git_filter_repo.py or did you forget to put the latter in your PYTHONPATH?")
+
+parser = argparse.ArgumentParser(
+          description='Offset a set of or all commits by n seconds, can be used to timeshift a repository')
+parser.add_argument('--offset', type=int,
+        help=("Offset in seconds"))
+parser.add_argument('--from-commit', type=str,
+        help=("Commit to start from"))
+parser.add_argument('--to-commit', type=str,
+        help=("Commit to finish at"))
+args = parser.parse_args()
+
+print(args)
+
+shouldOffset = args.from_commit is None
+
+def offsetDate(timestampAndUtcOffset):
+  timestampAndUtcOffset = timestampAndUtcOffset.decode('utf-8')
```

Comment:
Why decode into utf-8?  The values are all numeric or a space.  Seems it'd be easier to just

timestamp, timezone = timestampAndUtcOffset.split(b' ')
return b'%d %s' % (int(timestamp)+args.offset, timezone)

Reply:
Best I could do as a complete newbie to Python :)